paddle/pserver/LightNetwork.cpp

```diff
@@ -384,19 +384,22 @@ void SocketClient::TcpClient(const std::string &serverAddr, int serverPort) {
 
   /// Now connect to the server
   int retry_second = 0;
-  int error = 0;
   do {
-    error = connect(sockfd, (sockaddr *)&serv_addr, sizeof(serv_addr));
-    if (error == ECONNREFUSED) {
+    if (connect(sockfd, (sockaddr *)&serv_addr, sizeof(serv_addr)) == 0) {
+      break;
+    }
+
+    if (errno == ECONNREFUSED) {
       LOG(WARNING) << "connection refused by pserver, try again!";
       if (retry_second++ >= 7) {
         LOG(FATAL) << "connection refused by pserver, maybe pserver failed!";
       }
       std::this_thread::sleep_for(std::chrono::seconds(1));
```

Comment:
It seems that `retry_second` is defined by never used here. Should we make this line the following?

```cpp
std::this_thread::sleep_fo(std::chrono::seconds(retry_seconds);
```